ocaml/xapi-idl/lib/observer_skeleton.ml

```diff
@@ -0,0 +1,48 @@
+(*
+ * Copyright (c) Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+[@@@ocaml.warning "-27"]
+
+let u x = raise Observer_helpers.(Observer_error (Errors.Unimplemented x))
+
+module Observer = struct
+  type context = unit
+
+  let create ctx ~dbg ~uuid ~name_label ~attributes ~endpoints ~enabled =
+    u "Observer.create"
+
+  let destroy ctx ~dbg ~uuid = u "Observer.destroy"
```

Comment:
Could use `u __FUNCTION__` but would prefer `unimplemented __FUNCTION__`

Reply:
Good idea, that's much cleaner. I'll create a ticket to do this as it's also all over storage_skeleton.ml too.
I presume you mean you'd like the `u` function to be renamed to `unimplemented`?